src/tasks/processBaseFieldsCopyTask.ts

```diff
@@ -0,0 +1,135 @@
+import https from 'https';
+import {
+	isprocessBaseFieldsCopyTaskJobPayload,
+	TaskStatus,
+	isWritableBaseFieldWithLocalizationsContextArray,
+	type WritableBaseFieldWithLocalizationsContextArray,
+} from '../types';
+import {
+	createOrUpdateBaseField,
+	createOrUpdateBaseFieldLocalization,
+	loadBaseFieldsCopyTask,
+	updateBaseFieldsCopyTask,
+} from '../database/operations';
+import type { JobHelpers } from 'graphile-worker';
+
+export const fetchBaseFieldsFromRemote = async (
+	synchronizationUrl: string,
+): Promise<WritableBaseFieldWithLocalizationsContextArray> =>
+	new Promise((resolve, reject) => {
+		https
+			.get(`${synchronizationUrl}/baseFields`, (res) => {
+				let data = '';
+				res.on('data', (chunk) => {
+					data += chunk;
+				});
+				res.on('end', () => {
+					try {
+						const parsedData = JSON.parse(data) as unknown;
+						if (!isWritableBaseFieldWithLocalizationsContextArray(parsedData)) {
+							throw new Error('Invalid data received from remote pdc instance');
+						}
+						resolve(parsedData);
+					} catch (err) {
+						reject(
```

Comment:
Will `reject` cause it to try again?  Do we want it to try again if the format is unexpected?

Reply:
My understanding is that on promise rejection, the copyBasefields function that calls the fetch intercepts the failure and updates the status of the task, which I believe means that it will not get called again (which I think is what we want, so long as it is logged!)